react/week4/github-users/src/UserContext.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from "react";
+
+export const UserContext = React.createContext();
+
+const UserProvider = ({ children }) => {
+  const [users, setUsers] = useState([]);
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(true);
+  const [searchInput, setSearchInput] = useState("");
+  const [count, setCount] = useState(0)
+
+  const fetchGitUsers = async (input) => {
+    try {
+      await fetch(`https://api.github.com/search/users?q=${input}`)
+        .then((response) => {
+            if (!response.ok) {
+                throw Error("Could not fetch the data ");
+            } else {
+                return response.json();
+            }
+        })
+        .then((data) => {
+          setUsers(data.items)
+          setCount(data.total_count)
+          setLoading(false)
+        });
+    } catch (error) {
+        setError(error.message)
+    }
+};
+
+useEffect(() => {
+    if (!searchInput || error) {
+        setLoading(false);
+    } else {
+        setLoading(true);
+        fetchGitUsers(searchInput);
+    }  
+  }, [searchInput, error]);
+
+  return (
+    <UserContext.Provider
+      value={{ users, searchInput, setSearchInput, loading, count, error }}
+    >
+      {children}
+    </UserContext.Provider>
+  );
+};
+
```

Comment:
As mentioned before about creating your custom hook. Here I would export the following

`export const useUser = useContext(UserContext)`

That way in other files I can simply `import { useUser } from './UserContext'`

and use in the component as such `const { values } = useUser()`

You very first custom hook!

This is definitely a nice to have and what you have done is great already but thought you may like the input!